submissions/the-darkesttt/frogger-task/app.js

```diff
@@ -0,0 +1,323 @@
+
+        this.menuDescriprion.innerText = 'Choose your character!';
+        this.menuList = document.createElement('ul');
+
+        playerSkins.forEach((item) => {
+            this.menuItem = document.createElement('li');
+    
+            const itemImg = document.createElement('img');
+            const imgSrc = item.src;
+            const imgId = item.id;
+            itemImg.setAttribute('src', imgSrc);
+            itemImg.setAttribute('id', imgId);
+            const imgDesc = document.createElement('p');
+            imgDesc.innerText = imgId;
+    
+            this.menuList.appendChild(this.menuItem);
+            this.menuItem.appendChild(itemImg);
+            this.menuItem.appendChild(imgDesc);
+            
+        });
+
+        document.body.appendChild(this.bg);
+        document.body.appendChild(this.menu);
+
+        this.menu.appendChild(this.menuTitle);
+        this.menu.appendChild(this.menuDescriprion);
+        this.menu.appendChild(this.menuList);
+
+        this.menuList.addEventListener('click', (event) => {
+            const listTarget = event.target;
+            if (listTarget.tagName == "UL") return;
+        
+            player.sprite = listTarget.getAttribute("src");
+    
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+        });
+    }
+
+    death(points) {
+        if (this.menuList.parentElement != null) {
+            this.menuList.parentElement.removeChild(this.menuList);
+        }
+
+        this.bg.classList.remove('menu-animation');
+        this.menu.classList.remove('menu-animation');
+
+        this.menuDescriprion.innerText = 'You died! Your score:';
+
+        this.menuScore = document.createElement('span');
+        this.menuScore.innerText = points;
+        this.menuScore.classList.add('menu-score');
+
+        this.menu.appendChild(this.menuScore);
+
+        this.menuClose = document.createElement('button');
+        this.menuClose.innerText = 'Close menu and continue';
+        this.menuClose.classList.add('menu-close');
+        this.menu.appendChild(this.menuClose);
+
+        this.menuClose.addEventListener('click', (event) => {
+            this.bg.classList.add('menu-animation');
+            this.menu.classList.add('menu-animation');
+
+            this.menuDescriprion.innerText = '';
+            this.menuScore.parentElement.removeChild(this.menuScore);
+            this.menuClose.parentElement.removeChild(this.menuClose);
+        });
+    }
+}
+
+
+const startMenu = new Menu();
+startMenu.start();
+
+let points = 0;
+const score = document.createElement('p');
+score.classList.add('score');
+document.body.appendChild(score);
+
+function updateScore() {
+    score.innerHTML = 'Score: ' + points;
+}
+
+updateScore();
+
+function animateScoreGreen() {
+    score.classList.add('green');
+    setTimeout(function() {
+        score.classList.remove('green');
+    }, 1000)
+}
+
+function animateScoreBlue() {
```

Comment:
Functions `animateScoreBlue` and `animeateScoreGreen` are identical, the only thing changing here is the color. You can write a neat tool for yourself here to highlight the scoreboard with some color by creating a function that would receive a color and a timeout value and just reuse it. Like 
```js
function hightlightScore(color, duration) { ...
```